### PR TITLE
Use a fixed number of commits when calculating flaky percentage

### PR DESCRIPTION
**Pull request by keyonghan**
This fixes part of https://github.com/flutter/flutter/issues/92502.

Instead of querying build stats in past two weeks, this PR updates to query recent 100 (default value) builds for each builder.

### CODE REVIEW

File: app_dart/lib/src/service/bigquery.dart [hunk 1]
```diff
@@ -35,14 +37,15 @@ select builder_name,
...
-where date>=date_sub(current_date(), interval 14 day) and
-      date<=current_date() and
+from (select *, row_number() over (partition by builder_name order by time desc) as rank from `flutter-dashboard.datasite.luci_prod_build_status`)
+where date>=date_sub(current_date(), interval 30 day) and
       builder_name not like '%Drone' and
       repo='flutter' and
       branch='master' and
       pool = 'luci.flutter.prod' and
-      builder_name not like '%Beta%'
+      builder_name not like '%Beta%' and
```
Comment (CaseyHillers) [inline app_dart/lib/src/service/bigquery.dart:46]: What about stable?

Comment (keyonghan) [reply app_dart/lib/src/service/bigquery.dart:46]: Stable and dev do not run on `master` branch (we do not have any data under both `master` and `stable/dev`).

Comment (CaseyHillers) [reply app_dart/lib/src/service/bigquery.dart:46]: Out of curiosity, why does beta run?

Comment (keyonghan) [reply app_dart/lib/src/service/bigquery.dart:46]: @godofredoc may have more context.
BTW: here is the beta executions under `master`: go/flutter_beta_runs.

Comment (godofredoc) [reply app_dart/lib/src/service/bigquery.dart:46]: Flakiness should be calculated only on master/main. Flakiness on release branches is not actionable as only approved cherry picks are allowed to land on those branches.

Comment (CaseyHillers) [reply app_dart/lib/src/service/bigquery.dart:46]: This sounds like a bug in how we're saving the data (branch isn't being set correctly)

**Review by CaseyHillers — APPROVED** [review]
Everything else LGTM

File: app_dart/lib/src/service/bigquery.dart (hunk 1 above)
Comment (keyonghan) [reply app_dart/lib/src/service/bigquery.dart:46]: Created: https://github.com/flutter/flutter/issues/93557

**Pull request merged**